site/pages/api/publish.api.ts

```diff
@@ -0,0 +1,179 @@
+import { Progress, SourceType } from "@create-disruptions-data/shared-ts/enums";
+import {
+    isEnvironmentReason,
+    isMiscellaneousReason,
+    isPersonnelReason,
+    PtSituationElement,
+    Reason,
+} from "@create-disruptions-data/shared-ts/siriTypes";
+import dayjs from "dayjs";
+import { NextApiRequest, NextApiResponse } from "next";
+import { parseCookies } from "nookies";
+import { randomUUID } from "crypto";
+import { COOKIES_CONSEQUENCE_INFO, COOKIES_DISRUPTION_INFO, CREATE_DISRUPTION_PAGE_PATH } from "../../constants";
+import { insertPublishedDisruptionIntoDynamo } from "../../data/dynamo";
+import { consequenceSchema } from "../../schemas/consequence.schema";
+import { createDisruptionSchema } from "../../schemas/create-disruption.schema";
+import { cleardownCookies, redirectTo, redirectToError } from "../../utils/apiUtils";
+import { getDatetimeFromDateAndTime } from "../../utils/dates";
+
+const publish = async (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+        const { [COOKIES_DISRUPTION_INFO]: disruptionInfo, [COOKIES_CONSEQUENCE_INFO]: consequenceInfo } = parseCookies(
+            { req },
+        );
+
+        const parsedDisruptionInfo = createDisruptionSchema.safeParse(JSON.parse(disruptionInfo));
+        const parsedConsequenceInfo = consequenceSchema.safeParse(JSON.parse(consequenceInfo));
+
+        if (!parsedDisruptionInfo.success || !parsedConsequenceInfo.success) {
+            redirectTo(res, CREATE_DISRUPTION_PAGE_PATH);
+            return;
```

Comment:
Do we need a TODO here over how we're going to handle this issue? I wonder if we should be throwing something here, long term.

Reply:
I don't know if there's much that would be worth doing in this situation, if a user has gone through the journey then this should never happen so it's really only to catch requests that aren't going via the expected route. I can add a warn in at least so we could see in the logs if it's happening.

Reply:
Yeah, that makes me think we should be throwing an error so at least the user knows something didnt go as planned, and can contact helpdesk.

Reply:
I've updated it to throw an error which will take them to the error page which mentions how to contact the help desk